src/main/java/com/nutomic/syncthingandroid/fragments/SettingsFragment.java

```diff
@@ -297,6 +316,22 @@ public boolean onPreferenceChange(Preference preference, Object o) {
         return true;
     }
 
+    @Nullable
+    private String formatWifiNameList(Set<String> ssids) {
+        if (ssids != null) {
+            ssids = new TreeSet<>(ssids);
+            StringBuilder sb = new StringBuilder();
+            for (String ssid : ssids) {
+                if (sb.length() > 0) {
+                    sb.append(", ");
+                }
+                sb.append(ssid.replaceFirst("^\"", "").replaceFirst("\"$", ""));
+            }
+            return sb.toString();
+        }
+        return null;
```

Comment:
You can probably shorten this with [TextUtils#join()](http://developer.android.com/reference/android/text/TextUtils.html#join%28java.lang.CharSequence,%20java.lang.Iterable%29). Also, I'd prefer to use empty lists instead of null (needs less error checking).